### PR TITLE
Add plugin logging plus fix double plugin startup

### PR DESCRIPTION
**Pull request by ptuytsch**

### CODE REVIEW

File: src/gateway/sensor_controller.py [hunk 1]
```diff
@@ -88,6 +89,7 @@ def _plugin_callback_action(self, source, action, action_payload):
                 self._publish_config()
             return sensor_dto
         if action == SensorEnums.CallbackAction.REPORT_STATUS:
+            logger.debug('Sensor registers status based on payload: {}'.format(action_payload))
```
Comment (LnL7) [inline src/gateway/sensor_controller.py:92]: Using "register" here is somewhat confusing

File: src/gateway/watchdog.py [hunk 2]
```diff
@@ -101,6 +101,7 @@ def _controller_health(self, name, controller, device_reset):
                 device_reset()
                 if reset_action == 'service':
                     self._update_controller.block_updates()
+                    logger.error('Master communication issues, restarting service...')
```
Comment (LnL7) [inline src/gateway/watchdog.py:104]: This doesn't only trigger for master communication, the name/controller can be either master or energy.

File: src/plugin_runtime/runtime.py [hunk 3]
```diff
@@ -33,6 +34,33 @@
...
+        if self.file_logger is None and self.enabled:
+            self.file_logger = open(filename, 'a')
+
+    def log(self, obj, msg):
+        if self.enabled and self.file_logger is not None:
+            self.file_logger.write("{} [runtime: {}]: {}\n".format(datetime.datetime.now().replace(microsecond=0).isoformat(), obj, msg))
+            self.file_logger.flush()
+
+
+runtime_logger = RuntimeLogger(enabled=False)
```
Comment (LnL7) [inline src/plugin_runtime/runtime.py:54]: What is this for? Looks like functionality that's covered by the native logging library.

File: src/gateway/watchdog.py (hunk 2 above)
Comment (ptuytsch) [reply src/gateway/watchdog.py]: This was indeed to debug another issue, Was not needed in this context. Is removed.

File: src/logs.py [hunk 4]
```diff
@@ -14,13 +14,15 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import logging
-import sys
 from logging import handlers, Handler
 import os
 import re
-import constants
+from six.moves.configparser import ConfigParser, NoOptionError, NoSectionError
```
Comment (LnL7) [inline src/logs.py:20]: With the python2 environment 3rd party modules won't be available until `System.import_libs()` is called. Do all the entrypoints do this before anything related to logging is setup and imported?

**Review by LnL7 — APPROVED** [review]

File: src/logs.py (hunk 4 above)
Comment (ptuytsch) [reply src/logs.py:20]: Had to check, but yes this is one of the first things that happens on the plugin runtine as well. This does also work on a python2 running gateway.

**Pull request merged**